S26-documentation/12-non-breaking-space.t

```diff
@@ -1,163 +1,86 @@
-    ];
-
-    # make one testing row per non-breaking space char
-    for @nbchars -> $nbc {
-        # first column
-        $code ~= "Raku";
-        $code ~= $nbc.chr;
-        $code ~= "Language";
-        for @bchars -> $c {
-            $code ~= $c.chr;
-        }
-        $code ~= "is the best!";
-
-        # second column
-        $code ~= ' | ';
-        $code ~= "Raku ";
-        $code ~= $nbc.chr;
-        $code ~= " Language";
-        for @bchars -> $c {
-            $code ~= $c.chr;
-        }
-        $code ~= "is the best!\n";
-    }
-    $code ~= "=end table\n";
-
-    my $n = +@nbchars;
-    # the planned num of tests is 1 + $n
-    $code ~= "plan 1 + $n * 2;\n";
-    $code ~= "my \$n = $n;\n";
-    $code ~= "my \$SPACE = ' ';\n";
-
-    $code ~= q:to/HERE/;
-        my $r = $=pod[0];
-        is $r.contents.elems, $n, "table has $n elements (rows)";
-        HERE
-
-    # create separate sub-tests for each non-breaking space char
-    for 0..^$n -> $i {
-        $code ~= "\n\{\n";
-        $code ~= "my \$i = $i;\n";
-        $code ~= "my \$row = $i + 1;\n";
-        $code ~= "my \$nbspc = \"{@nbchars[$i].chr}\";\n";
-
-        $code ~= q:to/HERE/;
-            my $res0 = "Raku{$nbspc}Language is the best!";
-            my $res1 = "Raku {$nbspc} Language is the best!";
-            # show cell chars separated by pipes
-            =begin comment
-            my $c0 = $r.contents[$i][0].comb.join('|');
-            my $r0 = $res0.comb.join('|');
-            my $c1 = $r.contents[$i][1].comb.join('|');
-            my $r1 = $res1.comb.join('|');
-            =end comment
-            my $c0 = show-space-chars($r.contents[$i][0]);
-            my $r0 = show-space-chars($res0);
-            my $c1 = show-space-chars($r.contents[$i][1]);
-            my $r1 = show-space-chars($res1);
-
-            is $r.contents[$i][0], $res0, "row $row, col 1: '$c0' vs '$r0'";
-            is $r.contents[$i][1], $res1, "row $row, col 2: '$c1' vs '$r1'";
-            HERE
-
-        $code ~= "}\n";
-    }
-
-    $code ~= q:to/HERE/;
-        ##### some subroutines for the EVALed file #####
-        sub int2hexstr($int, :$plain) {
-            # given an int, convert it to a hex string
-            if !$plain {
-                return sprintf("0x%04X", $int);
-            }
-            else {
-                return sprintf("%X", $int);
-            }
-        }
+use lib $?FILE.IO.parent(2).add('packages/Test-Helpers/lib');
+use Test::Misc :int2hexstr, :show-space-chars;
+
+plan 9;
+
+# two-column table with various whitespace chars
+=begin table
+Raku Language	  ᠎            　is the best! | Raku   Language	  ᠎            　is the best!
+Raku Language	  ᠎            　is the best! | Raku   Language	  ᠎            　is the best!
+Raku⁠Language	  ᠎            　is the best! | Raku ⁠ Language	  ᠎            　is the best!
+Raku﻿Language	  ᠎            　is the best! | Raku ﻿ Language	  ᠎            　is the best!
+=end table
```

Comment:
I think removal of the pod-generation part is unfortunate. Hard-coding of whitespaces directly into the test code is dangerous because a "malicious" editor could "normalize" some of them. The best solution, to my view, is to use the approach I've shown in https://github.com/Raku/roast/issues/657#issuecomment-664522912 where pod code is `EVAL`ed but then the Pod::Block object is pulled from inside of the `EVAL` scope.